nodejs/src/index.js

```diff
@@ -7,34 +7,43 @@ var Compression = exports.Compression = {
 	Gzip: 1
 };
 
-exports.write = function (struct, compression, callback) {
-	switch (compression || Compression.None) {
-		case Compression.None:
-			serialize(struct, callback);
-			break;
-		case Compression.Gzip:
-			zip(struct, callback);
-			break;
-		default:
-			callback(new Error('Unable to understand the compression applied'));
-			break;
-	}
+exports.write = function (struct, compression, callback, noSettings) {
+    if (!noSettings) {
```

Comment:
I can just add the Compression.Disable for the write function and avoid passing in any extra arguments. In the read function if the noSettings argument in the third place, the new version won't be compatible with the old one.